### DISCUSSION TITLE
NextCloud Self-hosted Free Users Limit

### DISCUSSION THREAD

**Discussion by thiagoor-cpu** [nextcloud/all-in-one#5109] [Questions] [answered]
Hello,

I have a simple question, but I couldn't find an official answer anywhere. The NextCloud Administration Manual lists several installation options (VM Image, Snap, AIO, Docker, Pi) and the "source"/from scratch manual option. I understand that the soft and hard limits of users depend on the hardware, and not the installation option. However, on this [NextCloud AIO page](https://nextcloud.com/all-in-one/), it mentions, "Nextcloud offers an All-in-one Docker and VM image and direct deployments in a number of cloud environments like Azure and AWS," and states that the limit for the "community" version is 100 users. Some forum users also mentioned a "soft" limit of 100 users in some .ini file.

The "all-in-one" language on the page is confusing. Is it only referring to the Docker and VM versions, or does it include all versions (manual, Snap, Pi, AIO)? According to the Administration Manual, all installation options except AIO (VM Image, Snap, Docker, Pi) are "community" versions.

My question is: What is the user limit for the free self-hosted versions (VM Image, Snap, AIO, Docker, Pi, from scratch)? Does this limit apply only to users created outside of LDAP, or does it also apply when using LDAP?

My use case involves NextCloud AIO with an external LDAP, and I have 150 users. I want to know if this setup will work.

Thanks!

**Comment by szaimen** [comment, marked as answer]
Hi, just to leave this here, the limits can be removed by purchasing an enterprise license for AIO: https://nextcloud.com/all-in-one/

**Reply by thiagoor-cpu** [reply]
Ok I understand I can purchase a license, but what is the current limit and in which versions it apply? "Nextcloud AIO", "VM", "Docker", "Pi", "from scratch/manual installation", Snap???? With LDAP or without LDAP creating manually the users?

Please, I need a little more of information.

**Reply by robsoncombr** [reply]
Question remains right? Are there limits for the other distro types but all-in-one?

**Reply by denis-ismailaj** [reply]
@szaimen why did you mark this as the answer (twice)? How is "you can remove the restrictions here" an answer to "what are the restrictions"?

**Comment by denis-ismailaj** [comment]
The user limitation has no legal basis and is misleading.

This repository claims to use the AGPL-3.0 license and ships with an unmodified version of the AGPL-3.0.

As such, users are free to do with this software anything that the AGPL-3.0 allows them to.

Section 7 of the license says:

> If the Program as you received it, or any part of it, contains a notice stating that it is governed by this License along with a term that is a further restriction, you may remove that term. 

If Nextcloud Gmbh wants to place further restrictions they need to write their own license and use another name that is not under the copyright of the Free Software Foundation.

For reference, see [this statement](https://www.fsf.org/blogs/licensing/protecting-free-software-against-confusing-additional-restrictions) by the FSF.

Nextcloud should rectify this issue if they want to be honest to their userbase.
